catshadow/client.go

```diff
@@ -612,6 +612,20 @@ func (c *Client) createContact(nickname string, sharedSecret []byte) error {
 	return err
 }
 
+func (c *Client) getConversation(nickname string) map[MessageID]*Message {
+	conversationsCopy := make(map[MessageID]*Message)
+	c.conversationsMutex.Lock()
+	conv, ok := c.conversations[nickname]
+	if !ok {
+		return nil
```

Comment:
@david415 we should unlock here as well right

Reply:
yes!!! good catch. terrible mistake to make. @amitbasuri thanks